### ISSUE TITLE
Shared Detections for URLs?

### ISSUE THREAD

**Issue by SethFalco** [qeeqbox/social-analyzer#28]
When a service is open-source and can be self-hosted, this often leads to it being self-hosted on many different domains.

Examples are like:
* Mastodon
  * `https://fosstodon.org/@{username}`
  * `https://mastodon.cloud/@{username}`
  * `https://mastodon.social/@{username}`
  * `https://mastodon.technology/@{username}`
  * `https://mastodon.xyz/@{username}`
  * `https://mstdn.io/@{username}`
* Discourse
  * `https://ask.fedoraproject.org/u/{username}`
  * `https://community.signalusers.org/u/{username}`
  * `https://community.cloudflare.com/u/{username}`
  * `https://community.bitwarden.com/u/{username}/summary` 
  * `https://discourse.wicg.io/u/{username}/summary`
  * `https://discuss.atom.io/u/{username}/summary`
  * `https://forum.leasehackr.com/u/{username}/summary`
  * `https://discuss.elastic.co/u/{username}`
  * `https://forum.sublimetext.com/u/{username}`
  * `https://forums.whonix.org/u/{username}`
* GitLab
  * `https://gitlab.com/{username}`
  * `https://gitlab.gnome.org/{username}`
* Weblate
  * `https://hosted.weblate.org/user/{username}/`
* MediaWiki
  * `https://wikipedia.org/wiki/User:{username}`
> Only including links that exist in the repository.

Most Mastodon instances can share detections for example, same with all Discourse instances.
The only time this may not be the case is if a particular service is using a very outdated version of the software which then it may have to be separated.

It might help reduce the maintenance effort and total size of `sites.json`, by adding a way that many domains can share the same detections rather than writing/copying the same detections multiple times?

**Comment by giga-a** [comment]
I added generic and shared detections into sites.json as list of entries https://github.com/qeeqbox/social-analyzer/commit/9525193934de615e85a825c2ed0329105e322c22. The generic detections could be used with all websites. And, the shared detection for the ones you mentioned.

The backend logic for generic and shared detection is being tested. It will be pushed to Github in stages..

**Comment by giga-a** [comment]
I have manually added the shared detections logics https://github.com/qeeqbox/social-analyzer/commit/96249fc42d4b025fdcc70ad0585a3dc20620b57e. Mastodon websites now have `"type":"shared"` & `"name":"mastodon"` in `detections`. This will make the engine checks in `shared_detections` for the `"name":"mastodon"` 🎉

Here is an example
```
node app.js --cli --mode "fast" --username "johndoe" --websites "fosstodon mastodon mstdn" --output "pretty"
[!] Detections are updated every often, make sure to get the most updated ones
[Good] NodeJS Version Check
[Checking] fosstodon.org
[Checking] mastodon.cloud
[Checking] mastodon.social
[Checking] mastodon.technology
[Checking] mastodon.xyz
[Checking] mstdn.io

[Detected] 3 Profile[s]

[
  {
    "found": 3,
    "link": "https://mastodon.cloud/@johndoe",
    "rate": "%100.00",
    "title": "johndoe (@johndoe@mastodon.cloud) - mastodon.cloud",
    "language": "English",
    "type": "unavailable"
  },
  {
    "found": 3,
    "link": "https://mastodon.social/@johndoe",
    "rate": "%100.00",
    "title": "johndoe (@johndoe@mastodon.social) - Mastodon",
    "language": "English",
    "type": "unavailable"
  },
  {
    "found": 3,
    "link": "https://mastodon.xyz/@johndoe",
    "rate": "%100.00",
    "title": "johndoe (@johndoe@mastodon.xyz) - Mastodon",
    "language": "English",
    "type": "unavailable"
  }
]

[Unknown] 3 Profile[s]

[
  {
    "link": "https://fosstodon.org/@johndoe",
    "title": "The page you are looking for isn't here.\n - Mastodon",
    "language": "English",
    "type": "unavailable"
  },
  {
    "link": "https://mastodon.technology/@johndoe",
    "title": "The page you are looking for isn't here.\n - Mastodon",
    "language": "English",
    "type": "unavailable"
  },
  {
    "link": "https://mstdn.io/@johndoe",
    "title": "The page you are looking for isn't here.\n - Mastodon",
    "language": "English",
    "type": "unavailable"
  }
]
```

**Comment by giga-a** [comment]
Hey, I finished making shared detection for the examples listed above!

**Comment by giga-a** [comment]
Added shared to the Python CLI 

```python
python3 app.py --cli --mode "fast" --username "johndoe" --websites "fosstodon mastodon mstdn" --output "pretty"
[!] Detections are updated very often, make sure to get the most up-to-date ones
[Checking] mastodon.cloud
[Checking] fosstodon.org
[Checking] mstdn.io
[Checking] mastodon.technology
[Checking] mastodon.xyz
[Checking] mastodon.social

[Detected] 3 Profile[s]

{
    "found": 3,
    "language": "English",
    "link": "https://mastodon.cloud/@johndoe",
    "rate": "%100.0",
    "title": "johndoe (@johndoe@mastodon.cloud) - mastodon.cloud",
    "type": "unavailable"
}

{
    "found": 3,
    "language": "English",
    "link": "https://mastodon.xyz/@johndoe",
    "rate": "%100.0",
    "title": "johndoe (@johndoe@mastodon.xyz) - Mastodon",
    "type": "unavailable"
}

{
    "found": 3,
    "language": "English",
    "link": "https://mastodon.social/@johndoe",
    "rate": "%100.0",
    "title": "johndoe (@johndoe@mastodon.social) - Mastodon",
    "type": "unavailable"
}


[unknown] 3 Profile[s]

{
    "language": "English",
    "link": "https://mastodon.technology/@johndoe",
    "title": "The page you are looking for isn't here. - Mastodon",
    "type": "unavailable"
}

{
    "language": "English",
    "link": "https://mstdn.io/@johndoe",
    "title": "The page you are looking for isn't here. - Mastodon",
    "type": "unavailable"
}

{
    "language": "English",
    "link": "https://fosstodon.org/@johndoe",
    "title": "The page you are looking for isn't here. - Mastodon",
    "type": "unavailable"
}
```
